### DISCUSSION TITLE
Autolayout Grid

### DISCUSSION THREAD

**Discussion by buhtz** [mwaskom/seaborn#3385] [Q&A]
To my understanding all grid classes derived from `Grid` are based on data. There purpose is not only to organize and layout multiple plots in a visual grid. Their purpose seems to be taking the relationship between subplot's data into account.

I need a grid with less features and without being coupled to data. I just need the layout feature without the need of importing something from `matplotlib`.

```
grid = AutoGrid(columns=2)

cmap_list = ['RdYlGn', 'bwr', 'PuOr', 'BrBG', 'PRGn', 'coolwarm']

for cmap in cmap_list:
    ax = create_sub_axes(mydata, cmap)
    grid.add(ax)

grid.figure.show()
```

Another variant would be: `grid = AutoGrid(rows=2)` or `grid = AutoGrid(columns=2, rows=2)`. The latter should raise exceptions if more than `columsn * rows` (`4`) axes are added to it. An alternative name would be `FlowGrid`.

**Comment by mwaskom** [comment]
From your example I have no idea what you're expecting the result to be but if you just want a grid of axes with no data, you should be using matplotlib directly.

**Reply by buhtz** [reply]
The result is a grid with 4 subplots in it. What data they show is not important to the grid. That is the point.
It is a Grid just to layout plots totally independent from the data.

And using something else is not an option. Exactly that is the problem.
I want to use seaborn and nothing else. Importing matplotlib, plotly etc are workarounds then. This mess up my code and make it harder to maintain.
